src/libraries/System.Net.Primitives/tests/UnitTests/CookieExceptionTest.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Net.Primitives.UnitTests.Tests;
+
+public sealed class CookieExceptionTest
+{
+    [Theory]
+    [InlineData("Testing the CookieException")]
+    [InlineData(null)]
+    public void Constructor_Message(string? message)
+    {
+        var exception = new CookieException(message);
+        Assert.Equal(message, exception.Message);
+    }
+
+    [Theory]
+    [InlineData("Testing the CookieException", true)]
+    [InlineData("Testing the CookieException", false)]
+    [InlineData(null, true)]
+    [InlineData(null, false)]
+    public void Constructor_Message_InnerException(string? message, bool innerException)
+    {
+        var inner = innerException ? new Exception() : null;
+        var exception = new CookieException(message, inner);
+
+        Assert.Equal(message, exception.Message);
+        Assert.Equal(inner, exception.InnerException);
```

Comment:
```suggestion
        Assert.Same(inner, exception.InnerException);
```

Reply:
There would be no difference in result, if I use Same.
`Exception` is an reference type with no `Equals` overload, so it already compares the reference.

https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK4wD4AEBMBGAWAChiA3CZAAgFMBHVCAGz0oF5KZqB3SgUQA8AxtQAOCKAHsYACgBEAQVkBKANxkKNekxxsO3PkNHipcxauLEseAJzSAJLN5bGISgG86DZm3aftAX2U1IitbBycvVzcAJWoAM2pkahhhCKYAZ2k/ZjhNLxwlQNUgA=